pkg/framework/util.go

```diff
@@ -297,6 +297,7 @@ func ListImage(c internalapi.ImageManagerService, filter *runtimeapi.ImageFilter
 
 // PullPublicImage pulls the public image named imageName.
 func PullPublicImage(c internalapi.ImageManagerService, imageName string, podConfig *runtimeapi.PodSandboxConfig) string {
+	imageName = TestContext.RegistryPrefix + imageName
```

Comment:
the imageName may already contains the image registry, e.g. `"gcr.io/cri-tools/test-image-1:latest"`. So adding another prefix may not work. I think we should parse the image first and replace the registry with `TestContext.RegistryPrefix`.

Reply:
I agree, I changed the implementation to parse the image name before. Since the parsing expects a fully qualified image name, we now fallback to the intial `imageName` if the parsing fails. If not, we split up the domain and replace it.